src/@next/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import { Icon } from '../Icon';
+import { Neutral } from '../utilities/colors';
+import {
+  ArrowsContainer,
+  CarouselContainer,
+  Indicator,
+  IndicatorsContainer,
+  ItemsContainer,
+  NextArrow,
+  PreviousArrow,
+  SlideContainer,
+} from './CarouselStyle';
+
+export type AspectRatio = {
+  width: number;
+  height: number;
+};
+
+export interface CarouselProps {
+  autoRotate?: boolean;
+  children: React.ReactNode;
+  /** i.e: 16:9 The first number represents the width, and the second number represents the height.
+   * `AspectRatio: { width: number, height: number }`
+   */
+  aspectRatio?: AspectRatio;
+  /** sets height of Carousel*/
+  height?: string;
+  /** sets width of Carousel */
+  width?: string;
+  indicatorPosition?: 'top' | 'bottom' | 'left' | 'right' | 'outer';
+  indicatorType?: 'dot' | 'line' | 'slider';
+  /** When to show arrows*/
+  showArrow?: 'always' | 'hover' | 'never';
+}
+
+export const Carousel = ({
+  autoRotate = false,
+  children,
+  aspectRatio,
+  height,
+  width,
+  indicatorPosition = 'outer',
+  indicatorType = 'line',
+}: CarouselProps) => {
+  const slideRef = useRef(null);
+  const [activeIndex, setActiveIndex] = useState(0);
+  const [hideArrows, setHideArrows] = useState(false);
+
+  const childrenList = React.Children.toArray(children).filter(child =>
+    React.isValidElement(child)
+  );
+  const childrenLength = childrenList.length;
+
+  const updateActiveIndex = useCallback(
+    (index: number) => {
+      if (index < 0) {
+        setActiveIndex(childrenLength - 1);
+        return;
+      }
+
+      if (index === childrenLength) {
+        setActiveIndex(0);
+        return;
+      }
+
+      setActiveIndex(index);
+    },
+    [childrenLength]
+  );
+
+  const handlePreviousClick = () => {
+    updateActiveIndex(activeIndex - 1);
+  };
+
+  const handleNextClick = () => {
+    updateActiveIndex(activeIndex + 1);
+  };
+
+  const handleMouseEnter = () => {
+    if (autoRotate) {
+      setHideArrows(false);
+      return;
+    }
+  };
+
+  const handleMouseLeave = () => {
+    if (autoRotate) {
+      setHideArrows(true);
+      return;
+    }
+  };
+
+  useEffect(() => {
+    if (autoRotate) {
+      setTimeout(() => {
+        updateActiveIndex(activeIndex + 1);
+      }, 3000);
+    }
+  }, [activeIndex, autoRotate, updateActiveIndex]);
+
+  return (
+    <CarouselContainer
+      aspectRatio={
+        aspectRatio ? `${aspectRatio?.width}/${aspectRatio?.height}` : null
+      }
+      height={height}
+      width={width}
+      onMouseEnter={handleMouseEnter}
+      onMouseLeave={handleMouseLeave}
+    >
+      <SlideContainer
```

Comment:
@ninariccimarie the slider doesn't seem to be responsive[as desired](https://nimb.ws/BmPxwt) with the indicators staying in the middle and left/right arrow remaining fixed on the two ends